src/js/utils/MesosStateUtil.js

```diff
@@ -2,8 +2,34 @@ import Util from './Util';
 
 const RESOURCE_KEYS = ['cpus', 'disk', 'mem'];
 
+function setSchedulerName(name, tasks) {
+  return tasks.map(function (task) {
+    let newTask = Object.assign({}, task);
+
+    newTask.scheduler = name;
+
+    return newTask;
+  });
+}
+
 const MesosStateUtil = {
 
+  assignSchedulerNameToTasks(state) {
```

Comment:
I mentioned in [my comment](https://github.com/dcos/dcos-ui/pull/1021#r78254227), we do have the scheduler (framework) information on the tasks. If you look at tasks and look for `framework_id`, that should be enough information for you to tell whether this scheduled from a marathon instance or not (by a look up of the framework in Mesos data):
<img src="https://cl.ly/1c3E291L052S/Image%202016-09-16%20at%2009.14.46.png" width="25%" />

We do a similar thing here: https://github.com/dcos/dcos-ui/blob/master/src/js/pages/task-details/TaskDetailsTab.js#L41

However, thinking a bit more about this, it will probably cause some issues when we do this for all tasks (which could be _many_).

So what I suggest is that we add another property, exactly as you are doing here, but in stead tells us what we need to know:
e.g. `isStartedByMarathon = framework.name === 'marathon'` for tasks that have been started by the root Marathon framework.

Since the framework name `marathon` have been reserved for the root Marathon, this should work.

This solution is, however still far from ideal (because of the vague name matching). What really needs to happen is we need some type of label on the Framework level (or task level even) that will tell us which Marathon framework it was started from, if any.
That way we will be more robust towards the future, when we need to support different Marathons in the UI.

cc. @pyronicide How can we get the ball rolling on getting such a label?